### ISSUE TITLE
Exceptions not caught by (guard) inside sexp_eval_string

### ISSUE THREAD

**Issue by Oxyd** [ashinn/chibi-scheme#525]
Hi. The following code demonstrates an issue I've just encountered:

```C
#include <chibi/eval.h>
#include <stdio.h>

int
main() {
  sexp ctx;
  sexp_gc_var1(result);
  ctx = sexp_make_eval_context(NULL, NULL, NULL, 0, 0);
  sexp_load_standard_env(ctx, NULL, SEXP_SEVEN);

  sexp_gc_preserve1(ctx, result);
  result = sexp_eval_string(ctx,
                            "(import (scheme base)) "
                            "(guard (e (#t #t)) "
                            "  (error \"test\") "
                            "  #f)",
                            -1, NULL);
  if (sexp_booleanp(result))
    printf("Success: %s\n", result == SEXP_TRUE ? "#t" : "#f");
  else if (sexp_exceptionp(result))
    puts("Failure: exception");
  else
    puts("Big failure");

  sexp_gc_release1(ctx);
  sexp_destroy_context(ctx);
  return 0;
}
```

I expect it to print "Success: #t", but instead it prints "Failure: exception".

Inside the REPL, that same Scheme code seems to work just fine
```
> (import (scheme base)) (guard (e (#t #t)) (error "test") #f)
> #t
```

**Comment by ashinn** [comment]
That's not what I get - I get "Big failure."  This is because `sexp_eval_string` only evaluates the first sexp in the string, and the result of the import is undef.  This surprises people, so I'll change that behavior.

You can use `sexp_debug(ctx, "message: ", obj)` to print out an object to stderr.

You can and always should actually check what an exception is.  sexp_debug works for this, but there's also `sexp_print_exception(ctx, err, SEXP_FALSE)`.

**Comment by Oxyd** [comment]
Ah, it makes sense that `sexp_eval_string` would only evaluate one sexp. I guess the behaviour is fine, but it might be helpful to state that more explicitly in the documentation.

It's strange that you are getting different results than I am, though. If it's behaving the way it should, I'd expect us both to get the same results. I tested this with Chibi built from latest master (80bf401), by the way.

Also how would I go about doing what I tried to do, then? I figured I'd simply split it into two `sexp_eval_string`s, but I can't seem to be able to get `(import (scheme base))` to work in `sexp_eval_string` either:

```C
result = sexp_eval_string(ctx, "(import (scheme base))", -1, NULL);
sexp_debug(ctx, "import: ", result);
```
produces
```
import: {Exception #19 user "undefined variable" (base) #<procedure #f> #f}
```

Is it even possible to do an `import` in `sexp_eval_string` or am I trying to use the completely wrong thing here?
